App.ts

```diff
@@ -44,62 +49,56 @@ class App {
-    //   const id = crypto.randomBytes(16).toString("hex");
-    //   console.log(req.body);
-    //     var jsonObj = req.body;
-    //     jsonObj.listId = id;
-    //     this.Lists.model.create([jsonObj], (err) => {
-    //         if (err) {
-    //             console.log('object creation failed');
-    //         }
-    //     });
-    //     res.send('{"id":"' + id + '"}');
-    // });
-
-    // router.post('/app/list2/', (req, res) => {
-    //   const id = crypto.randomBytes(16).toString("hex");
-    //   console.log(req.body);
-    //     var jsonObj = req.body;
-    //     jsonObj.listId = id;
-    //     let doc = new this.Lists.model(jsonObj);
-    //     doc.save((err) => {
-    //        console.log('object creation failed');
-    //     });
-    //     res.send('{"id":"' + id + '"}');
-    // });
-
-    // router.get('/app/list/:listId', (req, res) => {
-    //     var id = req.params.listId;
-    //     console.log('Query single list with id: ' + id);
-    //     this.Tasks.retrieveTasksDetails(res, {listId: id});
-    // });
-
-    // router.get('/app/list/', (req, res) => {
-    //     console.log('Query All list');
-    //     this.Lists.retrieveAllLists(res);
-    // });
-
-    // router.get('/app/listcount', (req, res) => {
-    //   console.log('Query the number of list elements in db');
-    //   this.Lists.retrieveListCount(res);
-    // });
+    });
+    router.get('/app/items/', (req, res) => {
+      console.log('Query All items');
+      this.Items.retrieveAllItems(res);
+    });
+    router.get('/app/restaurantmanagers/', (req, res) => {
+      console.log('Query All Restaurant Managers');
+      this.RestaurantManagers.retrieveAllRestaurantManagers(res);
+    });
+    router.get('/app/restaurantowners/', (req, res) => {
+      console.log('Query All Restaurant Owners');
+      this.RestaurantOwners.retrieveAllRestaurantOwners(res);
+    });
+    router.post('/app/restaurantmanagers/create-manager', (req, res) => {
+      //TODO: add prechecks
+
+      //params from request body
+      let createManager: object = {
+        userID: crypto.randomUUID(),
+        password: req.body.password,
+        connectStatus: req.body.connectStatus,
+        //IRestaurantManagerModel
+        managerName: req.body.managerName,
+        restaurantOwnerID: req.body.restaurantOwnerID,
+        restuarantID: req.body.restaurantID
+      };
+
+      console.log("[App] Creating new restaurant manager with:" + JSON.stringify(createManager));
+      this.RestaurantManagers.createRestaurantManager(res, createManager)
+      
+
+    })
+
+    router.post('/app/restaurantmanagers/create-owner', (req, res) => {
+      //TODO: add prechecks
+
+      //params from request body
+      let createOwner: object = {
+        userID: crypto.randomUUID(),
+        password: req.body.password,
+        connectStatus: req.body.connectStatus,
+        //TODO: might have to convert req.body.restaurantList from JSON to RestaurantModel type list 
```

Comment:
Convert the Schema/Model to Object type

Reply:
note: we converted restaurantList to type string array